### PR TITLE
Using tagName JSX variable for MenuItem

### LINKED ISSUE #185: Support for client-side routing on href props
Many components (such as `Menu`) can take an href prop. This prop maps to the href value of an `<a>` tag. For many react projects that use client-side routing, anchor tags are replaced by `<Link>` (e.g. in [react-router](https://github.com/ReactTraining/react-router/blob/master/docs/API.md#link)) components that ensure the routing happens client-side. 

If the `<Link>` component is placed inside the text prop, everything behaves as expected, but React throws a warning for having nested `<a>` tags. 

Perhaps we could supply our own component to be used in place of the `<a>` tag in these situations.

### PR DESCRIPTION
**Pull request by barkermn01**
#### Changes proposed in this pull request:

the MenuItem now supports the property tagName which defaults to "a"

#### Reviewers should focus on:

./src/components/menu/menuItem.tsx

#### notes,

I'm on windows so I'm unable to build any of the testing stuff because your `yarn verify` setup is requiring access to `bash`... so someone is going to have to test this as I'm unable to.

```cmd
PS C:\*obsured*\blueprint> yarn verify
yarn run v1.10.1
$ npm-run-all -s compile dist:libs dist:apps -p test lint
$ lerna run compile
lerna info version 2.7.1
lerna info versioning independent
$ tsc -p ./src
$ tsc -p ./src
lerna ERR! compile Errored while running script in '@blueprintjs/icons'
lerna ERR! execute callback with error
lerna ERR! Error: Command failed: yarn run compile
lerna ERR! 'bash' is not recognized as an internal or external command,
lerna ERR! operable program or batch file.
lerna ERR! error Command failed with exit code 1.
lerna ERR! ERROR: "compile:css" exited with 1.
```

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @barkermn01! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/core/src/components/menu/menuItem.tsx [hunk 1]
```diff
@@ -70,6 +70,14 @@ export interface IMenuItemProps extends IActionProps, ILinkProps {
      * @default true
      */
     shouldDismissPopover?: boolean;
+
+    /**
+     * Name of the HTML tag that wraps the MenuItem.
+     *
+     * By default a `<a>` is used
```
Comment (giladgray) [inline packages/core/src/components/menu/menuItem.tsx]: remove this line, it's covered by `@default` below.

**Comment by giladgray** [comment]
@barkermn01 got a moment to finish up this PR please?

File: packages/core/src/components/menu/menuItem.tsx (hunk 1 above)
Comment (barkermn01) [reply packages/core/src/components/menu/menuItem.tsx]: Will do

File: packages/core/src/components/menu/menuItem.tsx [hunk 2]
```diff
@@ -78,6 +86,7 @@ export class MenuItem extends React.PureComponent<IMenuItemProps & React.AnchorH
         multiline: false,
         popoverProps: {},
         shouldDismissPopover: true,
+        tagName: "a",
```
[the remark this answers is not in the document]
Comment (barkermn01) [reply packages/core/src/components/menu/menuItem.tsx]: Done removed this one and left the one on line 107

**Comment by barkermn01** [comment]
Updated as per your requests should be ready to go.

**Comment by giladgray** [comment]
@barkermn01 looks good! will release shortly.

**Pull request merged**